app/uk/gov/hmrc/securemessage/services/exception/SecureMessageException.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.securemessage.services.exception
+
+trait SecureMessageException {
```

Comment:
i realised now, we need secure message exception to be a class and extend from java exception, so that in the controller we can handle the InternalSeevrError Htpp 500 with the last case _: Left(error: Exception) =>

Reply:
I have changed it to this. I don't see why we need to extend Exception
class SecureMessageException(val message: String, val throwable: Option[Throwable] = None)